courses.py

```diff
@@ -1,68 +1,89 @@
+
+"""
+Assumptions:
+    All subject names are 4-letter acronyms
+    All course numbers are 4-digit numbers
+    If a new subject is listed, the format will be ABCD1234 (i.e. no spaces between name and number)
+    If a subject number is listed without a course subject acronym, it's subject is the most recently seen subject:
+        (ex. 'ABCD1000 2000' is equivalent to 'ABCD1000 ABCD2000')
+"""
+
+
+def parse_courses():
+    """
+    Parses the COURSES string into a dictionary.
+    :return: A dictionary whose keys are course name acronyms (ex: 'COMP')
+             and the values are arrays of course numbers (as a numeric string, ex: ['2500', '2501', '2510'])
+    """
+
+    subjects = {}
+    name = ''  # the most recent course name acronym (ex. 'COMP')
+
+    courses = re.sub(r'\([^)]*\)', '', COURSES).split()  # Remove parens and their contents
+
+    for course in courses:
+        if course == 'OR':
+            continue
+
+        if course[0].isalpha():
+            name = course[:4]
+            number = course[4:8]
+        else:
+            number = course[:4]
+
+        try:
+            subjects[name].append(number)
+        except KeyError:
+            subjects[name] = [number]
+
+    return subjects
+
+
+def lookup(subjects, target):
+    """
+    Get course names from the web and write them to the specified target file.
+
+    :param subjects: A dictionary containing course subject
+                     and number information, i.e. the return type of the above function.
+    """
+
+    # regex stuff
+    p = re.compile("<.*>.* - (.*)</TD>")
+    url = "https://wl11gp.neu.edu/udcprod8/bwckctlg.p_disp_course_detail?cat_term_in=201630&subj_code_in={0}&crse_numb_in={1}"
+
+    # Get progress bar ready.
+    toolbar_width = len(subjects)
+    sys.stdout.write("[%s]" % (" " * toolbar_width))
     sys.stdout.flush()
-  progress = not progress
```

Comment:
Not a huge fan of having each course be it's own tick. Reason being for very long lists of classes, the progress bar generally won't fit on a single line.